### PR TITLE
GEM-118 | add delete endpoint

### PR DESCRIPTION
**Pull request by pwgacek**

### CODE REVIEW

File: src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt [hunk 1]
```diff
@@ -255,4 +258,66 @@ class ExternalPaymentControllerIT(
...
+        // when
+        val response = service.delete(createGemUser(USER_ID, EMAIL), GROUP_ID, PAYMENT_ID)
+
+        // then
+        response shouldHaveHttpStatus OK
+        paymentRepository.findByPaymentIdAndGroupId(PAYMENT_ID, GROUP_ID).also {
+            it.shouldBeNull()
+        }
+    }
+    should("return forbidden if user is not a group member") {
```
Comment (lawos98) [inline src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt:359]: same question like PR before

File: src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt [hunk 2]
```diff
@@ -85,10 +87,26 @@ class PaymentService(
...
+
+    fun deletePayment(paymentId: String, groupId: String, userId: String) {
+        val paymentToDelete = paymentRepository.findByPaymentIdAndGroupId(paymentId, groupId) ?: throw MissingPaymentException(paymentId, groupId)
+
+        if (!userId.isCreator(paymentToDelete)) {
+            throw PaymentDeletionAccessException(userId, paymentId)
+        }
+
+        paymentRepository.delete(paymentToDelete)
+        archivedPaymentRepository.add(paymentToDelete)
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt:118]: should we just archive payments that was in statuses ACCEPTED

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt:118]: yes

File: src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt (hunk 1 above)
Comment (pwgacek) [reply src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt:359]: some response like PR before

File: src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt (hunk 2 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt:118]: ???

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt:118]: all expenses and pyaments should be deleteable. If not we should also disable editing accepted expense/payment cause then if u want to delete u just edit it then status is pending and deletion is enabled

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt:118]: So in situation when you make an error in creating expense (for example forgot about one of the participants) and other accept it (so status is accepted) you can not change it or delete it

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt:118]: i don't see any cons of deleting accepted expense/payment. It is decision of creator and only the creator would not get their money back)

**Review by lawos98 — APPROVED** [review]

**Pull request merged**